configure.ml

```diff
@@ -80,16 +80,46 @@ let expand start finish input output =
     exit 1;
   end
 
+let output_file filename lines =
+  let oc = open_out filename in
+  let lines = List.map (fun line -> line ^ "\n") lines in
+  List.iter (output_string oc) lines;
+  close_out oc
+
+let cc c_program =
+  let c_file = Filename.temp_file "configure" ".c" in
+  let o_file = c_file ^ ".o" in
+  output_file c_file c_program;
+  let found = Sys.command (Printf.sprintf "cc -Werror -c %s -o %s %s" c_file o_file "2>/dev/null") = 0 in
+  if Sys.file_exists c_file then Sys.remove c_file;
+  if Sys.file_exists o_file then Sys.remove o_file;
+  found
+
+let check_domain_create_has_config_param () =
+  let c_program = [
+    "#include <stdlib.h>";
+    "#include <xenctrl.h>";
+    "#include <xenguest.h>";
+    "int main(int argc, const char *argv){";
```

Comment:
In this case I did not really care about the detail, I copied the code from xen-api-lowlevel-libs. Do you think it is worth to cleanup the configure.ml and polish the c also there?